test/setup-tests.js

```diff
@@ -0,0 +1,66 @@
+import fs from 'fs';
+import path from 'path';
+import colors from 'colors/safe';
+
+global.window.app = {
+  mcApiUrl: 'http://localhost:8080',
+};
+
+// setup file
+const logOrThrow = (log, method, messages) => {
+  const warning = `console.${method} calls not allowed in tests`;
+  if (process.env.CI) {
+    log(...messages);
+    throw new Error(warning);
+  } else {
+    log(colors.bgYellow.black(' WARN '), warning, '\n', ...messages);
+  }
+};
+
+// eslint-disable-next-line no-console
+const logMessage = console.log;
+global.console.log = (...messages) => {
+  logOrThrow(logMessage, 'log', messages);
+};
+
+// eslint-disable-next-line no-console
+const logInfo = console.info;
+global.console.info = (...messages) => {
+  logOrThrow(logInfo, 'info', messages);
+};
+
+// eslint-disable-next-line no-console
+const logWarning = console.warn;
+global.console.warn = (...messages) => {
+  logOrThrow(logWarning, 'warn', messages);
+};
+
+// eslint-disable-next-line no-console
+const logError = console.error;
+global.console.error = (...messages) => {
+  logOrThrow(logError, 'error', messages);
+};
+
+// Avoid unhandled promise rejections from going unnoticed
+// https://github.com/facebook/jest/issues/3251#issuecomment-299183885
+// In Node v7 unhandled promise rejections will terminate the process
```

Comment:
I'm not sure we need this for UI-kit

Reply:
It's debatable. I'd like to ensure our tests fail in case of an unhandled rejection. I'm not sure if we need this setup for it or if there is another way.

If we decide to take this approach, we definitely need a script on CI which checks for the existence of `fail-tests-because-there-was-an-unhandled-rejection.lock` and fails CI in case it exists. Good catch 👍